### PR TITLE
Implement AWAY

### PR DESCRIPTION
**Pull request by progval**

### CODE REVIEW

File: sable_ircd/src/command/handlers/away.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+use super::*;
+use event::*;
+
+#[sable_macros::command_handler("AWAY")]
```
Comment (spb) [inline sable_ircd/src/command/handlers/away.rs:4]: The qualified name should be unnecessary here as it's included in `super::*`

File: sable_network/src/network/event/details.rs [hunk 2]
```diff
@@ -174,26 +174,28 @@ EventDetails => {
...
+    struct ChannelRoleUpdate {
         pub data: Option<state::ChannelRole>,
     }
 
     #[target_type(UserId)]
-    struct UserLogin
-    {
+    struct UserAway {
+        /// Empty iff the user is back from being away
+        pub reason: String,
```
Comment (spb) [inline sable_network/src/network/event/details.rs:194]: Is there a reason this isn't Option<String>?

File: sable_network/src/network/state/user.rs [hunk 3]
```diff
@@ -30,6 +30,8 @@ pub struct User {
     pub visible_host: Hostname,
     pub realname: String,
 
+    /// Empty iff the user is not away
+    pub away_reason: String,
```
Comment (spb) [inline sable_network/src/network/state/user.rs:34]: As above - why not Option?

File: sable_network/src/network/event/details.rs (hunk 2 above)
Comment (progval) [reply sable_network/src/network/event/details.rs:194]: To follow the "Make Illegal States Unrepresentable" principle, as `Some("")` would be an invalid state.

Comment (spb) [reply sable_network/src/network/event/details.rs:194]: Fair, but I feel that in this case "" is sufficiently semantically different from any other string value that it should be different in the type system. Perhaps `Option<NonEmptyString>` or something similar?

Comment (progval) [reply sable_network/src/network/event/details.rs:194]: We'd be wasting a whole machine word of memory!

Comment (progval) [reply sable_network/src/network/event/details.rs]: Don e

**Pull request merged**